src/hooks/useSyntheticsTest.ts

```diff
@@ -22,32 +22,77 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 */
 
-import { useCallback, useContext, useState } from "react";
-import { getCodeForResult, getCodeFromActions } from "../common/shared";
-import { ActionContext, Result } from "../common/types";
+import { IpcRendererEvent } from "electron";
+import {
+  useCallback,
+  useContext,
+  useEffect,
+  useReducer,
+  useState,
+} from "react";
+import { getCodeForFailedResult, getCodeFromActions } from "../common/shared";
+import { ActionContext, Result, TestEvent } from "../common/types";
 import { CommunicationContext } from "../contexts/CommunicationContext";
+import { ITestContext } from "../contexts/TestContext";
+import { resultReducer } from "../helpers/resultReducer";
 
-export function useSyntheticsTest(actions: ActionContext[][]) {
-  const [result, setResult] = useState<Result | undefined>(undefined);
+export function useSyntheticsTest(actions: ActionContext[][]): ITestContext {
+  const [result, dispatch] = useReducer(resultReducer, undefined);
   const [codeBlocks, setCodeBlocks] = useState("");
+  const [isTestRunning, setIsTestRunning] = useState(false);
   const { ipc } = useContext(CommunicationContext);
 
   const onTest = useCallback(
     async function () {
-      /**
-       * For the time being we are only running tests as inline.
-       */
       const code = await getCodeFromActions(ipc, actions, "inline");
-      const resultFromServer: Result = await ipc.callMain("run-journey", {
-        code,
-        isSuite: false,
-      });
+      if (!isTestRunning) {
+        // destroy stale state
+        dispatch({ data: undefined, event: "override" });
+        const onTestEvent = (_event: IpcRendererEvent, data: TestEvent) => {
+          dispatch(data);
+        };
 
-      setCodeBlocks(await getCodeForResult(ipc, actions, result?.journey));
-      setResult(resultFromServer);
+        ipc.on("test-event", onTestEvent);
+
+        try {
+          const promise = ipc.callMain("run-journey", {
+            code,
+            isSuite: false,
+          });
+          setIsTestRunning(true);
+          await promise;
+        } catch (e: unknown) {
+          // eslint-disable-next-line no-console
+          console.error(e);
```

Comment:
https://github.com/elastic/synthetics-recorder/pull/143/commits/a5d9674afdbb85c92aea24eca3a8b5cfb7244cd8

Reply:
I think the `logger` is unavailable on the front-end, its inclusion caused CI to fail.

Reply:
https://github.com/elastic/synthetics-recorder/pull/143/commits/639e23fdbd1c7826b1279eefc88165adef4acf2a

Reply:
@lucasfcosta I just realized we do have access to Electron services via `window.require` and the logger does work in this case.

Is there a reason to use the logger over `console.error` aside from removing the linting rule? I think it would be better to avoid the dependency complexity if possible, unless there's some kind of file logging that we'd be leveraging using `logger`. The way we import IPC is a good example of a pattern we'd want to use to implement this access to help keep our code testable, and surface area of coupling into these services to a minimum.